server/pulp/server/webservices/views/consumers.py

```diff
@@ -0,0 +1,1082 @@
+
+        link = add_link(consumer)
+
+        document = {
+            'consumer': consumer,
+            'certificate': certificate
+        }
+        response = generate_json_response_with_pulp_encoder(document)
+        redirect_response = generate_redirect_response(response, link['_href'])
+        return redirect_response
+
+
+class ConsumerResourceView(View):
+    """
+    View for single consumer.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, consumer_id):
+        """
+        Return a serialized object representing the requested consumer.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: id for the requested consumer
+        :type consumer_id: str
+
+        :return: Response containing data for the requested consumer
+        :rtype: django.http.HttpResponse
+        """
+
+        query_params = request.GET
+        manager = factory.consumer_manager()
+        consumer = manager.get_consumer(consumer_id)
+        consumer = expand_consumers(query_params, [consumer])[0]
+        add_link(consumer)
+        return generate_json_response_with_pulp_encoder(consumer)
+
+    @auth_required(authorization.DELETE)
+    def delete(self, request, consumer_id):
+        """
+        Delete a specified consumer.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: id for the requested consumer
+        :type consumer_id: str
+
+        :return: An empty response
+        :rtype: django.http.HttpResponse
+        """
+
+        manager = factory.consumer_manager()
+        response = manager.unregister(consumer_id)
+        return generate_json_response(response)
+
+    @auth_required(authorization.UPDATE)
+    @json_body_required
+    def put(self, request, consumer_id):
+        """
+        Update a specified consumer.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param consumer_id: id for the requested consumer
+        :type consumer_id: str
+
+        :return: Response representing the updated consumer
+        :rtype: django.http.HttpResponse
+        """
+
+        params = request.body_as_json
+        delta = params.get('delta')
+        manager = factory.consumer_manager()
+        consumer = manager.update(consumer_id, delta)
+        add_link(consumer)
+        return generate_json_response_with_pulp_encoder(consumer)
+
+
+class ConsumerBindingsView(View):
+    """
+    View for Consumer bindings - represents the collection of
+    objects used to associate a consumer and a repo-distributor
+    association.  Users wanting to create this association will
+    create an object in this collection.  Both bind and unbind
+    is idempotent.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, consumer_id, repo_id=None):
```

Comment:
good catch!
i'd prefer to file a bug because i think this is not that trivial issue and the team could have different opinions on how to fix that.
Indeed this issue was present even in webpy http://fpaste.org/204689/27715513/